lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1004,105 +1011,103 @@ export class UserAgentApplication {
-        // TODO: Do we need to check if cache result is empty before calling this?
-        if (!authenticationRequest.authorityInstance) {//Cache result can return null if cache is empty. In that case, set authority to default value if no authority is passed to the api.
-            authenticationRequest.authorityInstance = authority ? AuthorityFactory.CreateInstance(authority, this.validateAuthority) : this.authorityInstance;
+      // Cache result can return null if cache is empty. In that case, set authority to default value if no authority is passed to the api.
+      // TODO: Do we need to check if cache result is empty before calling this?
+      if (!authenticationRequest.authorityInstance) {//Cache result can return null if cache is empty. In that case, set authority to default value if no authority is passed to the api.
+          authenticationRequest.authorityInstance = authority ? AuthorityFactory.CreateInstance(authority, this.validateAuthority) : this.authorityInstance;
+      }
+      // cache miss
+      return authenticationRequest.authorityInstance.ResolveEndpointsAsync()
+      .then(() => {
+        // refresh attept with iframe
+        // Already renewing for this scope, callback when we get the token.
+        if (window.activeRenewals[scope]) {
+          this._logger.verbose("Renew token for scope: " + scope + " is in progress. Registering callback");
+          // Active renewals contains the state for each renewal.
+          this.registerCallback(window.activeRenewals[scope], scope, resolve, reject);
         }
-        // cache miss
-        return authenticationRequest.authorityInstance.ResolveEndpointsAsync()
-        .then(() => {
-          // refresh attept with iframe
-          // Already renewing for this scope, callback when we get the token.
-          if (window.activeRenewals[scope]) {
-            this._logger.verbose("Renew token for scope: " + scope + " is in progress. Registering callback");
-            // Active renewals contains the state for each renewal.
-            this.registerCallback(window.activeRenewals[scope], scope, resolve, reject);
-          }
-          else {
-            if (scopes && scopes.indexOf(this.clientId) > -1 && scopes.length === 1) {
-              // App uses idToken to send to api endpoints
-              // Default scope is tracked as clientId to store this token
-              this._logger.verbose("renewing idToken");
-              this.renewIdToken(scopes, resolve, reject, userObject, authenticationRequest, extraQueryParameters);
-            } else {
-              // renew access token
-              this._logger.verbose("renewing accesstoken");
-              this.renewToken(scopes, resolve, reject, userObject, authenticationRequest, extraQueryParameters);
-            }
+        else {
+          if (scopes && scopes.indexOf(this.clientId) > -1 && scopes.length === 1) {
```

Comment:
use array.includes

Reply:
.includes does not work on some browsers

Reply:
I am almost positive that is the benefit the typescript compiler gives us, going down to es5

Reply:
open to being wrong !

Reply:
Can you try branching and testing this? Happy to integrate it if that is the case.